identity-core/src/crypto/proof/jcs_ed25519.rs

```diff
@@ -54,18 +60,27 @@ impl<T> Verifier<T::Public> for JcsEd25519<T>
 where
   T: Verify,
 {
-  fn verify<X>(data: &X, signature: &SignatureValue, public: &T::Public) -> Result<()>
+  type AuthenticityError = VerificationError;
+  type SignatureVerificationError = VerificationError;
+
+  fn verify<X>(data: &X, signature: &SignatureValue, public: &T::Public) -> Result<(), VerificationError>
   where
     X: Serialize,
   {
-    let signature: &str = signature
-      .as_signature()
-      .ok_or(Error::InvalidProofValue("jcs ed25519"))?;
-
-    let signature: Vec<u8> = decode_b58(signature)?;
-    let message: Vec<u8> = data.to_jcs()?;
-
-    T::verify(&message, &signature, public)?;
+    let signature: &str = signature.as_signature().ok_or(ProofValueError("jcs ed25519"))?;
+
+    let signature: Vec<u8> =
+      decode_b58(signature).map_err(|_| VerificationProcessingError::from("unable to decode the signature"))?;
+    let message: Vec<u8> = data
+      .to_jcs()
+      .map_err(|_| VerificationProcessingError::from("unable to serialize input data"))?;
+
+    T::verify(&message, &signature, public).map_err(|err| match err.try_into() {
+      Ok(invalid_proof_value) => VerificationError::from(invalid_proof_value),
+      Err(_) => {
+        VerificationProcessingError::from("unable to verify the authenticity of the given data and signature").into()
+      }
+    })?;
```

Comment:
```suggestion
   T::verify(&message, &signature, public).map_err(|err| {
        let err: Result<ProofValueError, _> = err.try_into();
        match err {
          Ok(invalid_proof_value) => VerificationError::from(invalid_proof_value),
          Err(_) => {
            VerificationProcessingError::from("unable to verify the authenticity of the given data and signature").into()
          }
        }
      })?;
```

Nit: It would be great to annotate the type of what `err` is converted into, because both try_into and from are generic here. Makes it more readable.